cosmic-applet-power/src/main.rs

```diff
@@ -101,73 +130,139 @@ impl Application for Power {
                     destroy_popup(p)
                 } else {
                     self.id_ctr += 1;
-                    let new_id = window::Id::new(self.id_ctr);
+                    let new_id = window::Id(self.id_ctr);
                     self.popup.replace(new_id);
 
                     let mut popup_settings = self.applet_helper.get_popup_settings(
-                        window::Id::new(0),
+                        window::Id(0),
                         new_id,
                         None,
                         None,
                         None,
                     );
                     popup_settings.positioner.size_limits = Limits::NONE
-                        .min_width(100)
-                        .min_height(100)
-                        .max_height(400)
-                        .max_width(500);
+                        .min_width(100.0)
+                        .min_height(100.0)
+                        .max_height(400.0)
+                        .max_width(500.0);
                     get_popup(popup_settings)
                 }
             }
             Message::Settings => {
                 let _ = process::Command::new("cosmic-settings").spawn();
                 Command::none()
             }
-            Message::Lock => Command::perform(lock(), Message::Zbus),
-            Message::LogOut => Command::perform(log_out(), Message::Zbus),
-            Message::Suspend => Command::perform(suspend(), Message::Zbus),
-            Message::Restart => Command::perform(restart(), Message::Zbus),
-            Message::Shutdown => Command::perform(shutdown(), Message::Zbus),
+            Message::Action(action) => {
+                self.id_ctr += 1;
+                let id = window::Id(self.id_ctr);
+                self.action_to_confirm = Some((id, action));
+                return Command::batch(vec![
+                    Command::perform(async { sleep(Duration::from_secs(60)).await }, move |_| {
```

Comment:
We should probably just use tokio sleep https://docs.rs/tokio/latest/tokio/time/fn.sleep.html

Reply:
I think it needs to be async, else it would block, right?

Reply:
async is corrrect, but cant we just pass the async function, itself
```rust
                    Command::perform(sleep(Duration::from_secs(60), move |_| {
```

Reply:
Ok cool!